flink-connectors/flink-connector-kafka-0.10/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducer010.java

```diff
@@ -411,6 +414,18 @@ public void processElement(StreamRecord<T> element) throws Exception {
 		invokeInternal(element.getValue(), element.getTimestamp());
 	}
 
+	@Override
+	public void snapshotState(FunctionSnapshotContext context) throws Exception {
+		final FlinkKafkaProducerBase<T> internalProducer = (FlinkKafkaProducerBase<T>) userFunction;
+		internalProducer.snapshotState(context);
+	}
+
+	@Override
+	public void initializeState(FunctionInitializationContext context) throws Exception {
```

Comment:
nit: I would declare `initializeState` before `snapshotState`, just for the sake of a better logic flow.